src/main/java/bio/terra/service/resourcemanagement/flight/DeleteSentinelStep.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.service.resourcemanagement.flight;
+
+import bio.terra.model.BillingProfileModel;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.profile.flight.ProfileMapKeys;
+import bio.terra.service.resourcemanagement.azure.AzureMonitoringService;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteSentinelStep extends DefaultUndoStep {
+  private static final Logger logger = LoggerFactory.getLogger(DeleteSentinelStep.class);
+  private AzureStorageAccountService azureStorageAccountService;
+  private AzureMonitoringService monitoringService;
+
+  public DeleteSentinelStep(
+      AzureMonitoringService monitoringService,
+      AzureStorageAccountService azureStorageAccountService) {
+    this.monitoringService = monitoringService;
+    this.azureStorageAccountService = azureStorageAccountService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    FlightMap workingMap = context.getWorkingMap();
+    BillingProfileModel profileModel =
+        workingMap.get(ProfileMapKeys.PROFILE_MODEL, BillingProfileModel.class);
+    List<AzureStorageAccountResource> storageAccounts =
+        workingMap.get(
+            ProfileMapKeys.PROFILE_STORAGE_ACCOUNT_RESOURCE_LIST, new TypeReference<>() {});
+
+    for (AzureStorageAccountResource storageAccountResource : storageAccounts) {
+      var sentinel = monitoringService.getSentinel(profileModel, storageAccountResource);
+      if (sentinel != null) {
+        logger.info(
+            "Sentinel instance found for storage account {}; Attempting delete.",
+            storageAccountResource.getName());
+        monitoringService.deleteSentinelByStorageAccount(profileModel, storageAccountResource);
```

Comment:
Oh, no, I don't think it has to follow this pattern, but I do think it makes sense in this case. 

It does follow the pattern of the create step -- We check if the resource already exists before trying to create it. 
But, more than following this pattern, I think this is just an alternate to wrapping the delete in a try/catch. While this approach doubles the number of requests to Azure, it should help us debug issues with the delete. We'll know if we're attempting the delete then the resource should exist to be deleted. I think this is a fair trade because this won't be a heavily trafficked endpoint, but any information we can provide on it's success/failure will be helpful.

Reply:
Sorry I wasn't clear, I think the check before delete is fine (vs catch when delete fails), the part in particular I was wondering about is why you can't pass the result of `get()` to the `delete()` API. E.g., it looks like you could pass in the `id` from the object you get back:
```
        monitoringService.deleteSentinelById(profileModel, sentinel.id());
```

Also I wonder if the `get` should be baked into the `delete` or not; does every caller need to check before calling `delete`? the only caller I could find of `deleteSentinelById()` is in `CreateSentinelStep.undoStep()`, which doesn't call `get` first, but that looks like a bug, since a step's undo should be idempotent.